controllers/storagecluster/storageclasses.go

```diff
@@ -104,7 +104,7 @@ func (r *StorageClusterReconciler) createStorageClasses(sccs []StorageClassConfi
 		case (strings.Contains(sc.Name, "-ceph-rbd") || strings.Contains(sc.Provisioner, fmt.Sprintf("%s.rbd.csi.ceph.com", operatorNamespace))) && !scc.isClusterExternal:
 			// wait for CephBlockPool to be ready
 			cephBlockPool := cephv1.CephBlockPool{}
-			key := types.NamespacedName{Name: sc.Parameters["pool"], Namespace: sc.Parameters["clusterID"]}
+			key := types.NamespacedName{Name: sc.Parameters["pool"], Namespace: namespace}
```

Comment:
Is this change related to creating rados namespace?

Reply:
Yes, the `sc.Parameters["clusterID"]` will be updated with new value if rados namespace is passed.
So the actual namespace would be different

Reply:
What happens to existing StorageClasses? Will it affect upgrades?

Reply:
No it wont effect with upgrades, and with new clusters either one of it will get creating according to the paramters